### DISCUSSION TITLE
Joe & Mac - Ultimate Caveman Collection - SNES - Retro-Bit

### DISCUSSION THREAD

**Discussion by retrovik** [sanni/cartreader#822] [Need help]
Hello. I am trying to backup my copy of "Joe & Mac Ultimate Collection 3 in 1" from Retro-Bit. I am only able to read what seems to be the menu section. Is this type of cart currently supported? Thanks.

--------------------------------------

It is identified as this:

OSCR HW3 V12.0

[+] SNES/SFC (CLK0+1)

[+] SNES/SFC cartridge

Searching database...
Checksum: E39D

Title: WOODI SFC TEST
Revision: 0
Type: LoROM SlowROM
ICs: ROM ONLY
ROM Size: 256 KB (8 banks)
Save Size: 32 KB
Checksum: E39D
Press Button...

![IMG_7097](https://github.com/sanni/cartreader/assets/122476671/466cab90-0fa9-4c71-86c6-84516de0ed11)

![IMG_7095](https://github.com/sanni/cartreader/assets/122476671/3cd259d9-76f9-4ac7-b0a5-7fb7d05bf24d)

![IMG_7093](https://github.com/sanni/cartreader/assets/122476671/c1177b5b-0289-4e5e-ac76-c08573be9946)

**Comment by retrovik** [comment]
Just thought I would update this topic in case anyone else wants to dump this cart. 

I am able to dump the cart and extract working games of Joe & Mac 1 + 2 and Congo's Caper from the OSCR dump. Perhaps others with more knowledge of the SNES backup process or mappers may know more from these details.

I am using HW3 and 13.1 firmware:

Here is the log from my dump:

OSCR HW3 V13.1

[+] Super Nintendo/SFC

[+] SNES/SFC cartridge


Searching database...
Checksum: E39D

Title: WOODI SFC TEST
Revision: 0
Type: LoROM SlowROM
ICs: ROM ONLY
ROM Size: 256 KB (8 banks)
Save Size: 32 KB
Checksum: E39D
Press Button...

[+] Force cart type

[+] 4MB LoROM 256K SRAM

Press Button...

[+] Read ROM


Saving to SNES/ROM/LoROM/257/...
[*******************]
Checksum... F0C3 != E39D
Invalid Checksum
CRC32... B0F32EDD -> Not found

-------------------------------------------------

If you dump based on the OSCR automatic detection, you will only get a 256kb dump. This contains the menu (mostly), but there are glitches. It does not contain the games that the menu references.

Here are the steps to get a working (sort of) dump of the menu AND all three games:

You must set the SNES cart type to "4mb LoROM 256K SRAM" in the OSCR menu for SNES games. Dump the game. This "may" be a correct dump of the cart. I am not sure because I cannot test it on real hardware. However, all emulators I tried can load the image, show the menu, but will freeze when picking a game. Maybe no emulators support this setup yet or maybe the dump is incorrect in some way? I am not sure.

Here are the steps to extract each part from the 4 megabyte dump:

Open the image in your favorite hex editor. Extract the data at the following addresses:

1) The menu is at address 0x000000 to 0x0FFFFF (0x000000 to 0x7FFFF might actually be all you need though)

![01-Menu](https://github.com/sanni/cartreader/assets/122476671/9f762f6d-1132-49d7-a62e-a575e2a6b39e)

2) Joe & Mac is at address 0x100000 to 0x1FFFFF

![02-Joe and Mac](https://github.com/sanni/cartreader/assets/122476671/c76b733f-1fc6-4bd4-95b1-aec88687585c)


3) Congo's Caper is at address 0x200000 to 0x27FFFF

![03-Congo](https://github.com/sanni/cartreader/assets/122476671/dc098c75-6e72-4b2e-8334-a0bed6b3f5aa)


4) Joe & Mac 2 is at address 0x300000 to 0x3FFFFF


![04-Joe and Mac 2](https://github.com/sanni/cartreader/assets/122476671/10a4dc0e-cf4d-453a-87d7-2c4008b3c7f8)


Those shots are from Windows BSNES 115, None of the Retro-Bit games match known good CRC values for these games, but they seem to load and play fine. Maybe they were modified in some way by Retro-Bit.

This is great progress for me since I wanted to at least backup the games in the collection. However, it would also be great to have it working using the Retro-Bit menu as well. Anyway, just wanted to share these updates. I love all the great work done for the OSCR!

**Comment by fakkuyuu** [comment]
thanks

**Comment by jrz1333** [comment]
Hey @retrovik , 

I know this is a bit of an older topic and has probably been closed out. But did you ever find a way to get the menu to work fully with the games? I was trying to find a way last night to edit the hex codes of the game and look at the PRG ROM to see how it calls out everything and jumps, but I was confused on how everything was laid out. I believe you're on to something with them being modified by Retro-Bit, It could also be how the game data backs up with the OSCR. 

I might have found something for JMP call-outs, but I don't have the slightest idea where it's jumping to. Could be that I'm using MESEN in this case. As well as the address lines listed show up in the MESEN debugger as "unidentified".

**Reply by retrovik** [reply]
Sorry. I was never able to figure it out. I did test the backup on real hardware, but the menu did not work there either (to launch a game). The games all worked though if I started them directly.  Maybe there is some custom fpga / mapper type functionality on the Retro-Bit cart to handle the game menu. Or maybe the dump itself is still not correct. I am just not sure.

**Comment by jd213** [comment]
I was similarly able to extract Kiwame 3 and Biometal from the "16-bit Collection Athena Vol. 1" cart from Jnnex/Retro-Bit in Japan. Kiwame 3 was at 0x200000 to 0x27FFFF, and Biometal was at 0x300000 to 0x3FFFFF.  

The cart menu was at the same addresses as above, but there didn't appear to be a game in 0x100000 to 0x1FFFFF as it was almost all 00 values.  Also tried dumping the cart as a HiROM and ExROM, but that didn't seem to work either. I'm pretty illiterate when it comes to hex though.  Maybe the cart needs a different dumping method to extract all the games.
